src/osmdata-data_frame.cpp

```diff
@@ -0,0 +1,259 @@
+ *  Software Foundation, either version 3 of the License, or (at your option)
+ *  any later version.
+ *
+ *  osmdata is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ *  FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ *  details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  osmdata.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *  Author:     Mark Padgham 
+ *  E-Mail:     mark.padgham@email.com 
+ *
+ *  Description:    Modified version of 'osmdata-sf' to extract OSM data from an
+ *                  object of class XmlData and return it in Rcpp::List format,
+ *                  but in this case ignoring the actual geometric data. The
+ *                  returned object contains key-value data only, for return
+ *                  from the R function 'osmdata_data_frame'.
+ *
+ *  Limitations:
+ *
+ *  Dependencies:       none (rapidXML header included in osmdata)
+ *
+ *  Compiler Options:   -std=c++11
+ ***************************************************************************/
+
+#include "osmdata.h"
+
+#include <Rcpp.h>
+
+// Note: This code uses explicit index counters within most loops which use Rcpp
+// objects, because these otherwise require a 
+// static_cast <size_t> (std::distance (...)). This operation copies each
+// instance and can slow the loops down by several orders of magnitude!
+
+/************************************************************************
+ ************************************************************************
+ **                                                                    **
+ **          1. PRIMARY FUNCTIONS TO TRACE WAYS AND RELATIONS          **
+ **                                                                    **
+ ************************************************************************
+ ************************************************************************/
+
+
+//' get_osm_relations
+//'
+//' Return a dual Rcpp::List containing all OSM relations, the firmt element of
+//' which holds `multipolygon` relations, while the second holds all others,
+//' which are stored as `multilinestring` objects.
+//'
+//' @param rels Pointer to the vector of Relation objects
+//' @param nodes Pointer to the vector of node objects
+//' @param ways Pointer to the vector of way objects
+//' @param unique_vals Pointer to a UniqueVals object containing std::sets of all
+//'       unique IDs and keys for each kind of OSM object (nodes, ways, rels).
+//'
+//' @return A dual Rcpp::List, the first of which contains the multipolygon
+//'         relations; the second the multilinestring relations.
+//' 
+//' @noRd 
+Rcpp::DataFrame osm_df::get_osm_relations (const Relations &rels, 
+        const UniqueVals &unique_vals)
+{
+    std::vector <std::string> ids_mp, rel_id_mp; 
```

Comment:
```suggestion
    std::vector <std::string> ids_mp; 
```
Skip rel_id_mp var. As I understand, is only used to build the rownames and be substituted in the if statement